x/datadeal/keeper/deal.go

```diff
@@ -544,45 +550,82 @@ func (k Keeper) DeactivateDeal(ctx sdk.Context, msg *types.MsgDeactivateDeal) er
 		return sdkerrors.Wrapf(types.ErrDealDeactivate, "deal's status is not 'ACTIVE'")
 	}
 
-	deal.Status = types.DEAL_STATUS_INACTIVE
+	deal.Status = types.DEAL_STATUS_DEACTIVATING
+
+	err = k.SetDeal(ctx, deal)
+	if err != nil {
+		return sdkerrors.Wrapf(types.ErrDealDeactivate, err.Error())
+	}
+
+	params := k.oracleKeeper.GetParams(ctx)
+	VotingPeriod := params.VoteParams.VotingPeriod
+
+	deactivationHeight := ctx.BlockHeader().Height + DealDeactivationParam*int64(VotingPeriod/BlockPeriod)
```

Comment:
I think it would be good add more block height. (+1 or higher)
```suggestion
	deactivationHeight := ctx.BlockHeader().Height + DealDeactivationParam*int64(VotingPeriod/BlockPeriod) +1
```

Reply:
Okay, Thanks. It's good because it seems to have a minimum limit.